### ISSUE TITLE
Support custom order for <select> field

### ISSUE THREAD

**Issue by r-a-y** [wp-shortcake/shortcake#620]
Currently, for the `<select>` field, for the `options` parameter, if you pass a bunch of options with a custom key order:

```
array(
    '0'  => '---',
    '50' => 'Yes',
    '30' => 'No'
)
```

Shortcake sorts the options by key in ascending order, so the select dropdown field becomes:
-  `---`
- No
- Yes

It would be nice if Shortcake preserved the array key order as it was added.

**Comment by westonruter** [comment]
I've come across this issue recently of preserving the order of PHP-supplied `choices` as well. I'm not sure if Shortcake is sorting the choices itself, but objects (associative arrays) in JavaScript are not ordered like they are on PHP. So when PHP exports the array from the server to JavaScript, it should actually create pure array of JS objects that looks like:

``` json
[
    {"text": "---", "value": "0"},
    {"text": "Yes", "value": "60"},
    {"text": "No", "value": "30"}
]
```

Here's how I'm handling a `select` template with `choices` like this: https://github.com/xwp/wp-customize-posts/blob/b8324b5b1aed3b4b9ced435e03f856032a317764/php/class-wp-customize-dynamic-control.php#L124-L130

**Comment by mattheu** [comment]
I've run into this too - there's a PR here: https://github.com/wp-shortcake/shortcake/pull/591

We're not sorting this - its simply because object properties don't have an order. I think we should use an array of objects instead as this maintains the order.

Some backwards compatability handling is needed. The PR could do with some tests.

**Comment by goldenapples** [comment]
Resolved by #591.
